### DISCUSSION TITLE
taskbar group = pager not working (labwc 0.8.1)

### DISCUSSION THREAD

**Discussion by jonleivent** [LBCrion/sfwbar#279] [Q&A]
I have the pager working with labwc 0.8.1, but the `group = pager` setting for the taskbar just blanks the taskbar.  Is this the same issue as https://github.com/LBCrion/sfwbar/issues/240?

**Comment by LBCrion** [comment]
It's not quite the same reason,  but similar.  Ext-workspace and cosmic-workspaces protocols provide information on the available workspaces,  but don't provide toplevel to workspace mappings. I.e. sfwbar doesn't know which workspace each window belongs to. There are additional protocols planned to provide this information eventually,  but the work on these hasn't even started yet in wayland-protocols. (I think we'll need at least 3 new protocols).

#240 needs window geometry information,  which I don't think is expected to be available at all through the wayland protocols.

**Comment by Consolatis** [comment]
> but the work on these hasn't even started yet in wayland-protocols. (I think we'll need at least 3 new protocols)

Right, either 3 protocols ([-state](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/196), [-management](https://gitlab.freedesktop.org/wayland/wayland-protocols/-/merge_requests/156), -workspace_interop) or some new event for the wlr_foreign_toplevel_protocol supplying a `ext-workspace object` e.g. something like `workspace_entered(ext-workspace-handle)` and `workspace_left(ext-workspace-handle)`.
